src/cuda/beam_search_scorer_cuda.cpp

```diff
@@ -10,14 +10,14 @@
 
 namespace Generators {
 
-BeamSearchScorer_Cuda::BeamSearchScorer_Cuda(const GeneratorParams& parameters)
-    : stream_{GetStream()} {
+BeamSearchScorer_Cuda::BeamSearchScorer_Cuda(const GeneratorParams& parameters, std::span<int32_t> cuda_eos_tokens)
```

Comment:
Should we remove the `cuda` prefix?

```suggestion
BeamSearchScorer_Cuda::BeamSearchScorer_Cuda(const GeneratorParams& parameters, std::span<int32_t> eos_tokens)
```

Reply:
I added that on purpose to make it clear this was cuda memory, as it's not passed as a DeviceSpan.